### ISSUE TITLE
[iOS] Remotely triggered SKCanvasView.InvalidateSurface does not redraw

### ISSUE THREAD

**Issue by Caopi** [mono/SkiaSharp#375]
Background and implementation:

I am currently trying to implement a shared drawing experience using Xamarin.Forms on mobile platforms, i.e. including a page with a SkiaSharp canvas, Touch listeners and a Realtime connection.

The drawing itself is basically implemented like this: https://developer.xamarin.com/guides/xamarin-forms/advanced/skiasharp/paths/finger-paint/, sample project here: https://github.com/xamarin/xamarin-forms-samples/tree/master/SkiaSharpForms/SkiaSharpFormsDemos

Also I am sending Messages to create a new path / add points to a path / ending a path via peer to peer connection. 

Problem:

The messages are coming through and the paths are properly transmitted and created (also verified fully working on Android and UWP). The only thing that is not working is the redrawing trigger.

Details:

I am calling the exact same code, once triggered from my own drawing, once triggered via network message. This code adds points to my paths and then triggers the SKCanvasView.InvalidateSurface() method to trigger a redraw of the canvas. But, this exact part simply does not work on iOS when triggered remotely. Only when I draw another line locally and thus trigger the invalidation again, the view actually refreshes.

Basically it seems like there are some native checks in place (only on iOS?) that require some changes done via user input before it actually refreshes and seems to block me from triggering this via code. I have verified via debugging that the code actually executes the InvalidateSurface() method, but the redrawing function does not work.

Any ideas why this is happening? Is there some workaround like faking a change on the canvas?

**Comment by mattleibow** [comment]
I am not sure exactly, but you may be trying to draw on a background thread? You have to make sure you are on the UI/main thread when hitting the UI.

**Comment by Caopi** [comment]
That was actually it, thank you. Sometimes you miss the most obvious things.
